mongorestore/restore.go

```diff
@@ -335,10 +306,16 @@ func (restore *MongoRestore) RestoreIntent(intent *intents.Intent) Result {
 	return result
 }
 
-func convertLegacyIndexes(indexes []IndexDocument) {
+func (restore *MongoRestore) convertLegacyIndexes(indexes []IndexDocument) {
 	for _, index := range indexes {
-		convertLegacyIndexKeys(index)
-		convertLegacyIndexOptions(index)
+		bsonutil.ConvertLegacyIndexKeys(index.Key, index.Options["ns"].(string))
+
+		// It is preferable to use the ignoreUnknownIndexOptions on the createIndex command to
+		// force the server to remove unknown options. But ignoreUnknownIndexOptions was only added in 4.1.9.
```

Comment:
I don't think it should affect this PR, but just curious to get your thoughts @tfogo:
would it be reasonable to ask for a backport of ignoreUnknownIndexOptions?

Reply:
I thought about this and I don't think there's any point. We'll only get support in a patch release so we'll still have to support multiple pre 4.1.9 patches that don't have it. The logic would be the same.